### PR TITLE
include gulp-data pipe

### PR DESCRIPTION
**Pull request by ryanluton**
Included the file.data attr from the gulp-data plugin

https://github.com/colynb/gulp-data

Verified working correctly when passing json into my generator-gulp-webapp project using nunjucks like so. 

```
gulp.task('views', function () {
  $.nunjucksRender.nunjucks.configure(['app/']);
  return gulp.src('app/*.html')
    .pipe(data(function() {
      return require('./mock-data.json');
    }))
    .pipe($.nunjucksRender())
    .pipe(gulp.dest('.tmp'))
});
```

### CODE REVIEW

**Comment by carlosl** [comment]
Thanks Ryan! I think I'm going to comment in the line where it merges the data instead of overwrites it in case some people are using both methods. Let me know what you think before I accept the merge.

**Comment by ryanluton** [comment]
That sounds good to me, I had just personally wanted the implicit include on my project. 

Thanks for the initial work on the renderer 👍=

File: index.js [hunk 1]
```diff
@@ -12,6 +13,15 @@ module.exports = function (options) {
 			return cb();
 		}
 
+		if (file.data) {
+  			//options = file.data;
+  			//custom objext.index for prototype
+  			options = file.data.results[0];
```
Comment (at0g) [inline index.js]: Why is it looking for a results array on file.data? 

The gulp-data example does not work.

Eg - this **should** work with gulp data:

```
gulp.src('path/to/templates/**/*.nunj')
  .pipe( data( function(file){
    return { example: 'data for ' + file.relative };
  }) )
  .pipe( nunjucksRender() )
  .pipe( gulp.dest('path/to/output')
;
```

**Comment by ryanluton** [comment]
I had updated my personal fork of this to work with a project I had going on. If you take a look at   6cc31768d457d7c71513c01d7cdf3c49aa2085c6 this was my original pull-request. Sorry for the confusion.

**Comment by at0g** [comment]
@ryanluton Thanks for the reply - I see what's happened... 

Not sure if you're aware of this or not, but making a commit to an open PR updates that PR (allowing the contributor to make amends based on peer review without creating a new PR for every amend).

So basically the subsequent commits you made to your fork (17 days ago) after sending the PR got merged into the master branch (12 days ago) of this repo

Hopefully @carlosl will either revert to `6cc3176`, or merge #14

**Pull request merged**